plugins/fireblocks/fireblocks-signer.ts

```diff
@@ -0,0 +1,291 @@
+      this.vaultAccountId,
+      this._bridge.assetId
+    );
+    return ethers.utils.getAddress(addresses[0].address);
+  }
+
+  setNextTransactionMemo(memo: string): void {
+    this.memo = memo;
+  }
+
+  _fail(message: string, operation: string): Promise<any> {
+    return Promise.resolve().then(() => {
+      log.throwError(message, Logger.errors.UNSUPPORTED_OPERATION, {
+        operation: operation,
+      });
+    });
+  }
+
+  async signMessage(message: Bytes | string): Promise<string> {
+    const data = ((typeof(message) === "string") ? toUtf8Bytes(message): message);
+    return await this._signMessage(data);
+  }
+
+  async _signMessage(message: Bytes): Promise<string> {
+    const txInfo = await this._bridge.sendRawTransaction(
+      keccak256(message).substring(2),
+      `Message signing request: ${this.memo}`
+    );
+    // There isn't really a transaction hash in raw signing but this
+    // does tell us when the request status has settled in fireblocks.
+    await this._bridge.waitForTxHash(txInfo.id);
+    const txDetail = await this.fireblocksApiClient.getTransactionById(
+      txInfo.id
+    );
+    if (txDetail.status !== TransactionStatus.COMPLETED) {
+      log.debug(txDetail);
+      throw new Error(`Transaction failed: ${JSON.stringify(txDetail)}`);
+    }
+    const sig = await txDetail.signedMessages![0].signature;
+    return sig.fullSig;
+  }
+
+  async _populateTransaction(
+    transaction: Deferrable<TransactionRequest>,
+    type: number = 2
+  ): Promise<UnsignedTransaction> {
+    // This is just hardcoded to 25 gwei in the JsonRPCProvider implementation :(
+    // const feeData = await this.provider?.getFeeData();
+    const feeData = getGasPriceSettings(this._bridge.getChainId());
+    let gasEstimate;
+    try {
+      gasEstimate = await this.provider?.estimateGas(transaction);
+    } catch (e: any) {
+      console.log('Error estimating gas', e.code);
+      gasEstimate = BigNumber.from(3000000); // 3M fallback (dangerous, improve this!)
+    }
+    const gasLimit = transaction.gasLimit !== undefined
+        ? BigNumber.from(await transaction.gasLimit)
+        : gasEstimate?.add(100000);
+    const maxPriority =
+      transaction.maxPriorityFeePerGas !== undefined
+        ? ethers.utils.parseUnits(`${transaction.maxPriorityFeePerGas!}`, 'gwei')
+        : ethers.utils.parseUnits(`${feeData?.maxPriorityFeePerGas}`, 'gwei') || undefined;
+    const maxFee =
+      transaction.maxFeePerGas !== undefined
+        ? ethers.utils.parseUnits(`${await transaction.maxFeePerGas!}`, 'gwei')
+        : ethers.utils.parseUnits(`${feeData?.maxFeePerGas!}`, 'gwei') || undefined;
+    const tx = await this.populateTransaction({
+      type,
+      ...transaction,
+      gasLimit,
+      ...(type === 2 && {
+        maxFeePerGas: maxFee,
+        maxPriorityFeePerGas: maxPriority,
+      }),
+      ...(type !== 2 && {
+        gasPrice: maxFee!.add(maxPriority!),
+      }),
+    });
+    console.log(
```

Comment:
would be nice to see some log consistency (ethers.log, hre.log, console.log, etc)

Reply:
Yeah, I wasn't leaning on `hre.log` inside this package that I'm hoping to extract into a stand-alone plugin since that's our own addition.  Could still be more consistent .though, agreed.